internal/ble_scanner/scanner.go

```diff
@@ -0,0 +1,106 @@
+package blescanner
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	"log"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/12urenloop/gonny-the-station-chef/internal/db"
+	"github.com/go-ble/ble"
+	"github.com/go-ble/ble/linux"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const ZEUS_MAC_PREFIX = "5a:45:55:53"
+
+type Scanner struct {
+	db  *db.DB
+	ctx context.Context
+}
+
+type BatonData struct {
+	ManufacturId      uint16
+	UptimeMs          uint64
+	BatteryPercentage uint8
+}
+
+func New(db *db.DB) *Scanner {
+	d, err := linux.NewDevice()
+	if err != nil {
+		log.Fatalf("can't create new device: %v\n", err)
+	}
+
+	ble.SetDefaultDevice(d)
+	ctx := context.Background()
+
+	scanner := Scanner{
+		db:  db,
+		ctx: ctx,
+	}
+	return &scanner
+}
+
+func (S *Scanner) scanFilter(a ble.Advertisement) bool {
+	return strings.HasPrefix(a.Addr().String(), ZEUS_MAC_PREFIX)
```

Comment:
Are the MAC addresses guaranteed to be lowercase? I now see that ronny also does this wrong, but maybe we should lowercase the address first before comparing (since we also store lowercase MACs in the DB)